src/wallet/spend.cpp

```diff
@@ -1160,7 +1160,21 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
     if (!select_coins_res) {
         // 'SelectCoins' either returns a specific error message or, if empty, means a general "Insufficient funds".
         const bilingual_str& err = util::ErrorString(select_coins_res);
-        return util::Error{err.empty() ?_("Insufficient funds") : err};
+        if (!err.empty()) return util::Error{err};
+
+        // Check if we have enough balance but cannot cover the fees
+        CAmount available_balance = preset_inputs.total_amount + available_coins.GetTotalAmount();
+        // Note: Since SFFO reduces existing outputs to cover fees, recipients_sum should already be
+        // strictly less than available_balance when enabled. Still, as a sanity-check, skip when SFFO is enabled.
+        if (available_balance >= recipients_sum && !coin_selection_params.m_subtract_fee_outputs) {
```

Comment:
In 98b7f1061ff104cdffebe1c2968c9f829d1a72f9 "wallet: introduce "tx amount exceeds balance when fees are included" error"

I'm pretty sure that `!coin_selection_params.m_subtract_fee_outputs` is not necessary as it should not be possible for `SelectCoins` to fail but `available_balance >= recipients_sum` with SFFO.

Perhaps this could be made into an `Assume` inside the `if`?